Tests/MongoSwiftTests/APMTests.swift

```diff
@@ -0,0 +1,203 @@
+    var type: EventType { get }
+}
+
+extension CommandEvent: StreamableEvent {}
+
+extension SDAMEvent: StreamableEvent {}
+
+@available(macOS 10.15, *)
+final class APMTests: MongoSwiftTestCase {
+    func testClientFinishesCommandStreams() async throws {
+        let elg = MultiThreadedEventLoopGroup(numberOfThreads: 1)
+        defer { elg.syncShutdownOrFail() }
+        let client = try MongoClient.makeTestClient(eventLoopGroup: elg)
+
+        // Task to create a commandEventStream and should return when the client is closed
+        let eventTask = Task { () -> Bool in
+            for try await _ in client.commandEventStream() {}
+            return true
+        }
+        try await client.db("admin").runCommand(["ping": 1])
+        try await client.close()
+
+        // Ensure the task returned eventually
+        _ = try await eventTask.value
+    }
+
+    func testCommandStreamHasCorrectEvents() async throws {
+        let commandStr: [String] = ["ping", "ping", "drop", "drop", "endSessions", "endSessions"]
+        let eventTypes: [EventType] = [
+            .commandStartedEvent,
+            .commandSucceededEvent,
+            .commandStartedEvent,
+            .commandSucceededEvent,
+            .commandStartedEvent,
+            .commandSucceededEvent
+        ]
+        let clientTask = try await self.withTestNamespace { client, db, _ -> Task<Int, Error> in
+            let cmdTask = Task { () -> Int in
+                var i = 0
+                for try await event in client.commandEventStream() {
+                    expect(event.commandName).to(equal(commandStr[i]))
+                    expect(event.type).to(equal(eventTypes[i]))
+                    i += 1
+                }
+                return i
+            }
+
+            try await db.runCommand(["ping": 1])
+            return cmdTask
+        }
+        let numEvents = try await clientTask.value
+        expect(numEvents).to(equal(6))
+    }
+
+    func testCommandStreamBufferingPolicy() async throws {
+        let insertsDone = NIOAtomic<Bool>.makeAtomic(value: false)
+        let clientTask = try await self.withTestNamespace { client, _, coll -> Task<Int, Error> in
+            let cmdBufferTask = Task { () -> Int in
+                var i = 0
+                let stream = client.commandEventStream()
+                try await assertIsEventuallyTrue(description: "inserts done") {
+                    insertsDone.load()
+                }
+
+                for try await _ in stream {
+                    i += 1
+                }
+                return i
+            }
+            for _ in 1...60 { // 120 events
+                try await coll.insertOne(["hello": "world"])
+            }
+            insertsDone.store(true)
+            return cmdBufferTask
+        }
+        let numEventsBuffer = try await clientTask.value
+        // Cant check for exactly 100 because of potential load balancer events getting added in during loop
```

Comment:
just out of curiosity what are the extra load balancer command events you were seeing?